### PR TITLE
Change Scalar to DataVector in DefiniteIntegral

### LINKED ISSUE #187: definite_integral should take a DataVector, not a Scalar<DataVector>
I do not see why it takes a Scalar<DataVector>  (In fact it didn't in PR #47, but did in PR #51 which says it was recreated from PR #47, and it looks like we reviewed PR #47, but not PR #51. So do we need to see if there are any other differences between the two PRs?)

### PR DESCRIPTION
**Pull request by mar-celine**
Fixes #187

### CODE REVIEW

File: tests/Unit/Numerical/Spectral/Test_DefiniteIntegral.cpp [hunk 1]
```diff
@@ -13,14 +13,20 @@
 #include "Numerical/Spectral/LegendreGaussLobatto.hpp"
 #include "tests/Unit/TestHelpers.hpp"
 
+// GCC 4.7.3 triggers the following for the CHECK() macros.
```
Comment (kidder) [inline tests/Unit/Numerical/Spectral/Test_DefiniteIntegral.cpp]: Why did this need to come back after the change?  Which compiler triggers the warning? What version of Catch are you using?

File: src/Numerical/Spectral/DefiniteIntegral.hpp [hunk 2]
```diff
@@ -29,7 +29,7 @@ namespace lgl {
  * \returns the definite integral of `integrand` on the manifold.
  */
 template <size_t Dim>
-double definite_integral(const Scalar<DataVector>& integrand,
+double definite_integral(const DataVector& integrand,
```
Comment (kidder) [inline src/Numerical/Spectral/DefiniteIntegral.hpp:30]: Remove the include for Tensor

**Review by kidder — CHANGES_REQUESTED** [review]

File: src/Numerical/Spectral/DefiniteIntegral.cpp [hunk 3]
```diff
@@ -12,16 +12,16 @@
 namespace {
 
 template <size_t Dim>
-Scalar<DataVector> integrate_over_last_dimension(const DataVector& integrand,
-                                                 const Index<Dim>& extents) {
+DataVector integrate_over_last_dimension(const DataVector& integrand,
+                                         const Index<Dim>& extents) {
```
Comment (nilsdeppe) [inline src/Numerical/Spectral/DefiniteIntegral.cpp]: noexcept things

File: src/Numerical/Spectral/DefiniteIntegral.cpp [hunk 4]
```diff
@@ -30,31 +30,30 @@ namespace Basis {
...
-  return ddot_(num_points, weights.data(), 1, integrand.get().data(), 1);
+  return ddot_(num_points, weights.data(), 1, integrand.data(), 1);
 }
 
 }  // namespace lgl
 }  // namespace Basis
 
 /// \cond
-template double Basis::lgl::definite_integral<2>(const Scalar<DataVector>&,
+template double Basis::lgl::definite_integral<2>(const DataVector&,
```
Comment (nilsdeppe) [inline src/Numerical/Spectral/DefiniteIntegral.cpp]: noexcept

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

File: tests/Unit/Numerical/Spectral/Test_DefiniteIntegral.cpp (hunk 1 above)
Comment (nilsdeppe) [reply tests/Unit/Numerical/Spectral/Test_DefiniteIntegral.cpp]: `GCC 5 triggers the -Wparentheses warning for some CHECK() macros, sometimes...`

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

File: tests/Unit/Numerical/Spectral/Test_DefiniteIntegral.cpp [hunk 5]
```diff
@@ -13,14 +13,17 @@
 #include "Numerical/Spectral/LegendreGaussLobatto.hpp"
 #include "tests/Unit/TestHelpers.hpp"
 
+// GCC 5 triggers the -Wparentheses warning for some CHECK() macros,
```
Comment (nilsdeppe) [inline tests/Unit/Numerical/Spectral/Test_DefiniteIntegral.cpp]: Gabe, plz...

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Review by kidder — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**